### PR TITLE
DBZ-20 A subset of string, time and number datatypes

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSchemaChangeEventEmitter.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public void emitSchemaChangeEvent(Receiver receiver) throws InterruptedException
     private SchemaChangeEventType getSchemaChangeEventType() {
         switch(ddlLcr.getCommandType()) {
             case "CREATE TABLE": return SchemaChangeEventType.CREATE;
-            case "ALTER TABLE": throw new UnsupportedOperationException("ALTER TABLE not yet implemented");
+//            case "ALTER TABLE": throw new UnsupportedOperationException("ALTER TABLE not yet implemented");
```
Comment (gunnarmorling) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSchemaChangeEventEmitter.java]: Is that intentional?

File: debezium-core/src/test/java/io/debezium/data/SchemaAndValueField.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                return;
+            }
+        }
+        else {
+            Assertions.assertThat(actualValue.getClass()).as("Incorrect value type for " + fieldName).isEqualTo(value.getClass());
+        }
+
+        if (actualValue instanceof byte[]) {
+            Assertions.assertThat((byte[]) actualValue).as("Values don't match for " + fieldName).isEqualTo((byte[]) value);
+        } else if (actualValue instanceof Struct) {
```
Comment (gunnarmorling) [inline debezium-core/src/test/java/io/debezium/data/SchemaAndValueField.java]: Can you move `else` to the next line?

File: debezium-core/src/test/java/io/debezium/data/SchemaAndValueField.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            final Object actualValue = actualStruct.get(field.name());
+            Assertions.assertThat(actualValue).as("No value found for " + fieldName).isNotNull();
+            Assertions.assertThat(actualValue.getClass()).as("Incorrect value type for " + fieldName).isEqualTo(expectedValue.getClass());
+            if (actualValue instanceof byte[]) {
+                Assertions.assertThat(expectedValue).as("Array is not expected for " + fieldName).isInstanceOf(byte[].class);
+                Assertions.assertThat((byte[]) actualValue).as("Values don't match for " + fieldName).isEqualTo((byte[]) expectedValue);
+            }
+            else if (actualValue instanceof Struct) {
+                assertStruct((Struct)expectedValue, (Struct)actualValue);
+            }else {
```
Comment (gunnarmorling) [inline debezium-core/src/test/java/io/debezium/data/SchemaAndValueField.java]: `else` on new line please.

**Review by gunnarmorling — COMMENTED** [review]
@jpechane Looks great. Just a few comments inline.

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSchemaChangeEventEmitter.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSchemaChangeEventEmitter.java]: @gunnarmorling This a is a good question - I need it for `ALTER TABLE ADD SUPPLEMENTAL LOG DATA (ALL) COLUMNS`. As I suppose this is soething that is needed we should either handle this case or throw only `WARN` for all alters

Comment (gunnarmorling) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSchemaChangeEventEmitter.java]: Yes, at least a warning should be logged.

**Comment by gunnarmorling** [comment]
Applied. Thank you very much, @jpechane!

**Pull request merged**